source/frontend/src/hooks/useSearch.ts

```diff
@@ -85,56 +102,65 @@ export function useSearch<ISearchResult extends object, IFilter extends object>(
     [noResultsWarning],
   );
 
-  // update search results whenever new data comes back from API endpoints
-  useDeepCompareEffect(() => {
-    if (!filter) {
-      return;
-    }
-
-    async function callApi() {
+  const pageSize = state.pageSize;
+  const sort = state.sort;
+  const currentPage = state.currentPage;
+  const filter = state.filter;
+  const callApi = useCallback(
+    async (currentPage: number, filter: IFilter) => {
       try {
-        setLoading(true);
+        setState({ loading: true });
         dispatch(showLoading());
         const { data } = await searchFn(filter, sort, currentPage, pageSize);
         if (isMounted()) {
           setSearchOutput(data, pageSize);
-          setError(undefined);
+          setState({ error: undefined });
         }
       } catch (e) {
         if (isMounted()) {
           setSearchOutput(undefined, 0);
-          setError((e as Error) ?? new Error('Something went wrong. Please try again.'));
+          setState({ error: (e as Error) ?? new Error('Something went wrong. Please try again.') });
         }
       } finally {
-        setLoading(false);
+        setState({ loading: false });
         dispatch(hideLoading());
       }
-    }
+    },
+    [dispatch, isMounted, pageSize, searchFn, setSearchOutput, sort],
+  );
 
-    callApi();
-  }, [isMounted, filter, currentPage, pageSize, searchFn, sort, refreshIndex, setSearchOutput]);
+  // update search results whenever new data comes back from API endpoints
+  useDeepCompareEffect(() => {
+    if (!filter) {
+      return;
+    }
+    callApi(currentPage, filter);
+  }, [currentPage, filter, callApi]);
 
   // allow manual re-triggers of the API calls
-  const execute = () => {
+  const executeFn = useCallback(() => {
     setRefreshIndex(r => r + 1);
-  };
+  }, []);
```

Comment:
**nit:** I don't think we ever use manual re-triggers of the search call - I think we could try to remove this. Also if you notice from the old code, the `refreshIndex` was in the dependency array of **useDeepEffect** to force the effect to re-run when the execute function was called. I don't see a similar logic with the updated code (one more reason to remove it)